### PR TITLE
Sanitize all uploaded files

### LINKED ISSUE #63: Sanitize all filenames
Same error as in #48. Affected fields are `Human.photo`, `Human.cv_file` and `Thesis.thesis_file`

### PR DESCRIPTION
**Pull request by Zelvuska**
Fix #63

### CODE REVIEW

File: fuuk/people/models/course.py [hunk 1]
```diff
@@ -32,9 +29,7 @@ def __unicode__(self):
 
 
 def attachment_filename(instance, filename):
-    base, ext = os.path.splitext(force_text(filename))
-    basename = slugify(base) + ext
-    return 'files/courses/' + basename
+    return sanitize_filename(filename, 'files/courses/')
```
Comment (ziima) [inline fuuk/people/models/course.py]: Directory names shouldn't end up with `/`.

File: fuuk/people/models/utils.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from django.utils.encoding import force_text
+from django.utils.text import slugify
+
+
+def sanitize_filename(filename, path=''):
+    '''
+    Sanitize filename and create correct path.
+    '''
+    base, ext = os.path.splitext(force_text(filename))
+    basename = slugify(base) + ext
```
Comment (ziima) [inline fuuk/people/models/utils.py]: We should consider that unicode chars can be in extension as well.

File: fuuk/people/models/utils.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    if path and not path.endswith(os.path.sep):
+        path += os.path.sep
+    return (path or '') + basename
```
Comment (ziima) [inline fuuk/people/models/utils.py]: `os.path.join` should handle all this.

File: fuuk/people/tests/test_models.py [hunk 4]
```diff
@@ -98,3 +114,20 @@ def test_file_unicode(self):
...
+        # Test `cv_filename` function
+        self.assertEqual(cv_filename(sentinel.instance, 'foo.dat'), 'cv/foo.dat')
+
+
+class TestImageFilename(TestCase):
+    """
+    Test `image_filename` function.
+    """
+    def test_image_filename(self):
+        self.assertEqual(image_filename(sentinel.instance, 'foo.dat'), 'img/person/foo.dat')
```
Comment (ziima) [inline fuuk/people/tests/test_models.py]: We should ensure that these functions sanitizes the inputs. It doesn't have to be thorough, but at least one check.

File: fuuk/people/tests/test_models.py [hunk 5]
```diff
@@ -49,6 +50,26 @@ def test_name(self):
...
+        self.assertEqual(sanitize_filename('foo.bar.dat'), 'foobar.dat')
+        self.assertEqual(sanitize_filename('foo bar.dat'), 'foo-bar.dat')
+        self.assertEqual(sanitize_filename('ěščřž.dat'), 'escrz.dat')
+        self.assertEqual(sanitize_filename(u'ěščřž.dat'), 'escrz.dat')
+        self.assertEqual(sanitize_filename('.dat'), 'dat')
+
+    def test_path(self):
+        self.assertEqual(sanitize_filename('foo.dat'), 'foo.dat')
+        self.assertEqual(sanitize_filename('foo.dat', ''), 'foo.dat')
+        self.assertEqual(sanitize_filename('foo.dat', None), 'foo.dat')
```
Comment (ziima) [inline fuuk/people/tests/test_models.py]: I don't think we have to allow `path=None` since the default is `''`.

**Comment by Zelvuska** [comment]
Done

File: fuuk/people/models/utils.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+'''
+Common utils for models.
+'''
```
Comment (ziima) [inline fuuk/people/models/utils.py]: I'd keep this in `people.utils`. It doesn't have to be used only in models.

Comment (Zelvuska) [inline fuuk/people/models/utils.py]: Moved

**Pull request merged**